javascript/src/main/components/BasicCourseSearch/SelectSubject.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { Form } from "react-bootstrap";
+import { compareValues } from "main/utils/sortHelper"
+
+const SelectSubject = ({ subjects, _subject, setSubject}) => {
+
+    const handleSubjectOnChange = (event) => {
+        setSubject(event.target.value);
+    };
+
+    subjects.sort(compareValues("subjectCode"));
+
+    return (
+        <Form.Group controlId="BasicSearch.Subject">
+            <Form.Label>Subject Area</Form.Label>
+            <Form.Control as="select" onChange={handleSubjectOnChange} >
+                {subjects.map(function (object, i) {
```

Comment:
Going along with the previous case of an error in the subjects API call, we would probably want to do a check here to see if we actually have subjects.

Also a small nitpick, but can we change this to an arrow function?

Reply:
Let's address that in a later PR.   We really need this in place so that other teams can get started on work.

Reply:
Opened issue #162 to address the potential error cases.